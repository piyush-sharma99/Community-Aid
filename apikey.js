/*
 *  ClassName: apikey.js
 *
 *  Date: 28/03/2021
 *
 * @author Piyush Sharma, X17342356
 *
 * @reference https://reactnative.dev/docs/activityindicator
 * @reference https://www.udemy.com/course/react-native-the-practical-guide/
 * @reference https://www.youtube.com/watch?v=8zT6CYu0iYQ
 *
 */

export const firebaseConfig = {
  apiKey: "AIzaSyDGliQfMfKpXUllG5ZPyClJL6C8VgFKXo8",
  authDomain: "community-aid.firebaseapp.com",
  databaseURL:
    "https://community-aid-default-rtdb.europe-west1.firebasedatabase.app",
  projectId: "community-aid",
  storageBucket: "community-aid.appspot.com",
  messagingSenderId: "529896701339",
  appId: "1:529896701339:web:1c30369d39f8654b01353e",
  measurementId: "G-PPD28P0PC6",
};
