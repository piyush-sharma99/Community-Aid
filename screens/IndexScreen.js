import React from 'react';
import { StyleSheet, Text, View, TextInput, Button, TouchableOpacity, Image} from 'react-native';

const IndexScreen = props => {
    return(
        <View style={styles.screen} >
        
        <Image 
        style={styles.logo}
        source={require('../assets/Logo.png')}
        />

        <View>
        <TouchableOpacity style={styles.loginBtn} onPress = {() => {
            props.navigation.navigate({routeName: 'Login'});
          }}>
          <Text style={styles.loginText} >Log In</Text>
        </TouchableOpacity>

        <TouchableOpacity style={styles.signBtn} onPress = {() => {
            props.navigation.navigate({routeName: 'Signup'});
          }}>
          <Text style={styles.signText}>Sign Up</Text>
        </TouchableOpacity>
        </View>

        </View>
    );
};

IndexScreen.navigationOptions = {
headerTitle: 'Welcome',
headerTitleStyle: { alignSelf: 'center' },
headerStyle: {
backgroundColor: '#2E86C1'
},
headerTintColor:"white"
};

const styles = StyleSheet.create({
    screen: {
        width:'100%',
        height:'100%',
        flex: 1,
        backgroundColor: '#2E86C1',
        alignItems: 'center',
        justifyContent: 'center'
    },

    logo:{
        justifyContent:"center",
        marginTop:-230
      },

      inputViewEmail:{
        width:"80%",
        backgroundColor:"#FDFEFE",
        borderRadius:25,
        height:50,
        marginTop:30,
        marginBottom:20,
        justifyContent:"center",
        padding:20,
        elevation: 10,
        borderWidth: 4,
        borderColor: "#fb5b5a"
      },

      inputViewPassword:{
        width:"80%",
        backgroundColor:"#FDFEFE",
        borderRadius:25,
        height:50,
        marginBottom:20,
        marginTop:10,
        justifyContent:"center",
        padding:20,
        elevation: 10,
        borderWidth: 4,
        borderColor: "#fb5b5a"
      },
      inputText:{
        height:50,
        color:"black"
      },

      forgot:{
        color:"white",
        fontSize:11
      },

      loginText:{
        color:"white",
        fontSize:20
      },

      signText:{
        color:"white",
        fontSize:20
      },

      forgot:{
        color:"white",
        fontSize:15,
        marginBottom:5,
        marginTop:20
      },

      loginBtn:{
        width:200,
        backgroundColor:"#fb5b5a",
        borderRadius:25,
        height:50,
        alignItems:"center",
        justifyContent:"center",
        marginTop:20,
        marginBottom:5
      },

      signBtn:{
        width:200,
        backgroundColor:"#fb5b5a",
        borderRadius:25,
        height:50,
        alignItems:"center",
        justifyContent:"center",
        marginTop:20,
        marginBottom:10
      },
      
});

export default IndexScreen;