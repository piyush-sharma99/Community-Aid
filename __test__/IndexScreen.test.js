import React from "react";
import IndexScreen from "../screens/IndexScreen";
import { firebaseConfig } from "../apikey";
import { cleanup } from "@testing-library/react-native";
import renderer from "react-test-renderer";
import * as firebase from "firebase";

firebase.initializeApp(firebaseConfig);

afterEach(cleanup);

test("Index renders correctly", () => {
  const tree = renderer.create(<IndexScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
