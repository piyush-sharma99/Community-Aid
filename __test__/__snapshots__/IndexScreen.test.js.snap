// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": "100%",
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "height": "100%",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 200,
          "justifyContent": "center",
          "marginTop": -300,
          "width": 200,
        }
      }
    />
    <View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fb5b5a",
            "borderColor": "#fff",
            "borderRadius": 15,
            "borderWidth": 3,
            "elevation": 10,
            "height": 50,
            "justifyContent": "center",
            "marginBottom": 5,
            "marginTop": 30,
            "opacity": 1,
            "shadowOpacity": 1,
            "width": 300,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 20,
            }
          }
        >
          Log In
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fb5b5a",
            "borderColor": "#fff",
            "borderRadius": 15,
            "borderWidth": 3,
            "elevation": 10,
            "height": 50,
            "justifyContent": "center",
            "marginBottom": 10,
            "marginTop": 20,
            "opacity": 1,
            "shadowOpacity": 1,
            "width": 300,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 20,
            }
          }
        >
          Sign Up
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fb5b5a",
            "borderColor": "#fff",
            "borderRadius": 15,
            "borderWidth": 3,
            "elevation": 10,
            "height": 50,
            "justifyContent": "center",
            "marginBottom": 10,
            "marginTop": 20,
            "opacity": 1,
            "shadowOpacity": 1,
            "width": 300,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 20,
            }
          }
        >
          Learn more about us!
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "flexWrap": "wrap",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderRadius": 25,
            "height": 80,
            "justifyContent": "center",
            "margin": 10,
            "marginBottom": -120,
            "marginTop": 120,
            "opacity": 1,
            "shadowOffset": Object {
              "height": 3,
              "width": 3,
            },
            "shadowOpacity": 1,
            "width": "25%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "borderRadius": 25,
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text />
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 15,
                "fontWeight": "bold",
                "marginTop": 10,
                "textAlign": "center",
              }
            }
          >
            Facebook
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderRadius": 25,
            "height": 80,
            "justifyContent": "center",
            "margin": 10,
            "marginBottom": -120,
            "marginTop": 120,
            "opacity": 1,
            "shadowOffset": Object {
              "height": 3,
              "width": 3,
            },
            "shadowOpacity": 1,
            "width": "25%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "borderRadius": 25,
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text />
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 15,
                "fontWeight": "bold",
                "marginTop": 10,
                "textAlign": "center",
              }
            }
          >
            Instagram
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderRadius": 25,
            "height": 80,
            "justifyContent": "center",
            "margin": 10,
            "marginBottom": -120,
            "marginTop": 120,
            "opacity": 1,
            "shadowOffset": Object {
              "height": 3,
              "width": 3,
            },
            "shadowOpacity": 1,
            "width": "25%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "borderRadius": 25,
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text />
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 15,
                "fontWeight": "bold",
                "marginTop": 10,
                "textAlign": "center",
              }
            }
          >
            linkedin
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
